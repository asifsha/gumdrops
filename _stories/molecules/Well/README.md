The `<Well>` component can be used for alert messages or other user notifications.

If you pass in  `button` and a callback function, the component will call that function when a user clicks on the x.
