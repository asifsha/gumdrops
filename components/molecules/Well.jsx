import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Well = ({ text, context, button, callback, className, style }) => {
    const baseClass = 'gds-well';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${context}`]: context
    });

    const buttonClass = cx('gds-well__button', {
        [`gds-well__button--${context}`]: context
    });

    return (
        <div className={rootClass} style={style}>
            <p className="gds-well__text">{text}</p>
            {button && callback && <button className={buttonClass} onClick={callback} />}
        </div>
    );
};

Well.displayName = 'Well';

Well.defaultProps = {
    text: null,
    context: null,
    button: false,
    callback: null,
    className: '',
    style: {}
};

Well.propTypes = {
    text: PropTypes.string,
    /** success, info, warning, danger */
    context: PropTypes.string,
    button: PropTypes.bool,
    callback: PropTypes.func,
    className: PropTypes.string,
    style: PropTypes.object
};

export default Well;
