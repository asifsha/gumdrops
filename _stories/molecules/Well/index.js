import React from 'react';
import { boolean, text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import Well from '../../../components/molecules/Well';

const contextOptions = {
    'No Value': '',
    success: 'success',
    warning: 'warning',
    info: 'info',
    danger: 'danger'
};

const component = () => (
    <Well
        text={text('Label', 'Default Well')}
        context={optionalSelect('Context', contextOptions, '')}
        button={boolean('Button', false)}
        callback={action('well_closed')}
        className={text('ClassName', '')}
        style={object('Style', {})}
    />
);

export default [readme, component];
