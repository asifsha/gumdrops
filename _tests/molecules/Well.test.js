/* globals mount */
import React from 'react';
import Well from '../../components/molecules/Well';

const defaultProps = {
    text: 'Foo',
    context: 'primary',
    className: '',
    style: { width: 100 }
};

describe('Expect <Well>', () => {
    it('to render', () => {
        const wrapper = mount(<Well {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render with a button', () => {
        const props = {
            ...defaultProps,
            button: false,
            callback: () => {}
        };
        const wrapper = mount(<Well {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
