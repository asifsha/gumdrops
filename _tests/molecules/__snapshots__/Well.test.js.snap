// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Well> to render 1`] = `
<Well
  button={false}
  callback={null}
  className=""
  context="primary"
  style={
    Object {
      "width": 100,
    }
  }
  text="Foo"
>
  <div
    className="gds-well gds-well--primary"
    style={
      Object {
        "width": 100,
      }
    }
  >
    <p
      className="gds-well__text"
    >
      Foo
    </p>
  </div>
</Well>
`;

exports[`Expect <Well> to render with a button 1`] = `
<Well
  button={false}
  callback={[Function]}
  className=""
  context="primary"
  style={
    Object {
      "width": 100,
    }
  }
  text="Foo"
>
  <div
    className="gds-well gds-well--primary"
    style={
      Object {
        "width": 100,
      }
    }
  >
    <p
      className="gds-well__text"
    >
      Foo
    </p>
  </div>
</Well>
`;
